osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerSpectateButton.cs

```diff
@@ -46,10 +51,11 @@ private void onClick()
         }
 
         [BackgroundDependencyLoader]
-        private void load()
+        private void load(OsuConfigManager config)
         {
             operationInProgress = ongoingOperationTracker.InProgress.GetBoundCopy();
             operationInProgress.BindValueChanged(_ => updateState());
+            automaticallyDownload = config.GetBindable<bool>(OsuSetting.AutomaticallyDownloadMissingBeatmaps);
```

Comment:
Shouldn't this respond to changes of this setting? If it's initially disabled and the user enables it, it won't start the download.